src/types/token_stream.cpp

```diff
@@ -19,87 +19,198 @@ constexpr unsigned tabulator_length = 8;
 TokenIt TokenStream::emplace_back(TokenType type, char value)
 {
 	_tokens.emplace_back(type, Literal(std::string(1, value)));
+	_formatted = false;
 	return --_tokens.end();
 }
 
 TokenIt TokenStream::emplace_back(TokenType type, const Literal& literal)
 {
 	_tokens.emplace_back(type, literal);
+	_formatted = false;
 	return --_tokens.end();
 }
 
 TokenIt TokenStream::emplace_back(TokenType type, Literal&& literal)
 {
 	_tokens.emplace_back(type, std::move(literal));
+	_formatted = false;
 	return --_tokens.end();
 }
 
 TokenIt TokenStream::emplace(const TokenIt& before, TokenType type, char value)
 {
 	_tokens.emplace(before, type, Literal(std::string(1, value)));
 	auto output = before;
+	_formatted = false;
 	return --output;
 }
 
 TokenIt TokenStream::emplace(const TokenIt& before, TokenType type, const Literal& literal)
 {
 	_tokens.emplace(before, type, literal);
 	auto output = before;
+	_formatted = false;
 	return --output;
 }
 
 TokenIt TokenStream::emplace(const TokenIt& before, TokenType type, Literal&& literal)
 {
 	_tokens.emplace(before, type, std::move(literal));
 	auto output = before;
+	_formatted = false;
 	return --output;
 }
 
 TokenIt TokenStream::push_back(const Token& t)
 {
 	_tokens.push_back(t);
+	_formatted = false;
 	return --_tokens.end();
 }
 
 TokenIt TokenStream::push_back(Token&& t)
 {
 	_tokens.push_back(std::move(t));
+	_formatted = false;
 	return --_tokens.end();
 }
 
 TokenIt TokenStream::insert(TokenIt before, TokenType type, const Literal& literal)
 {
+	_formatted = false;
 	return _tokens.insert(before, Token(type, literal));
 }
 
 TokenIt TokenStream::insert(TokenIt before, TokenType type, Literal&& literal)
 {
+	_formatted = false;
 	return _tokens.insert(before, Token(type, std::move(literal)));
 }
 
 TokenIt TokenStream::erase(TokenIt element)
 {
+	_formatted = false;
 	return _tokens.erase(element);
 }
 
 TokenIt TokenStream::erase(TokenIt first, TokenIt last)
 {
+	_formatted = false;
 	return _tokens.erase(first, last);
 }
 
 void TokenStream::move_append(TokenStream* donor)
 {
 	_tokens.splice(_tokens.end(), donor->_tokens);
+	_formatted = false;
 }
 
-void TokenStream::move_append(TokenStream* donor, TokenIt before)
+void TokenStream::move_append(TokenIt before, TokenStream* donor)
```

Comment:
What I noticed recently is that we are mixing here camel case which is usually established in yaramod however there are these methods with snake casing. Not something crucial but it would be nice to stick to single naming style.

Reply:
I am not sure whether we also want to rename `TokenStream::emplace_back` to `TokenStream::emplaceBack`. Also `pushBack` looks a bit strange when underlying List has these methods named in snake case. I would probably have big troubles to get used to it. Do you mind if we keep it `emplace_back` and `push_back`?

But I renamed `move_append` to `moveAppend` because standard c++ containers have no such methods and therefore it should not cause us troubles to get used to camel case in this case.

Reply:
Also `swap_tokens` will be renamed to `swapTokens`.